### PR TITLE
Move `TransactionStart`/`TransactionEnd`/`SetVariable` into `LogicalPlan::Statement`

### PR DESCRIPTION
**Pull request by alamb**
# Which issue does this PR close?

Follow on to https://github.com/apache/arrow-datafusion/pull/5827 from @avantgardnerio 

related to https://github.com/apache/arrow-datafusion/issues/3349 

# Rationale for this change

As I have been working on documenting datafusion (see #5499), the number of distinct variants of LogicalPlan is confusing. As I was reviewing https://github.com/apache/arrow-datafusion/pull/5827 I realized it made the problem (slightly) worse and figured it was time to set us on a more sustainable course.

I believe we need to invest to help make the code base easier to understand and extend. 

# What changes are included in this PR?

1. Adding another layer of indirection in `LogicalPlan::Statement` and (start) gathering statements together

# Are these changes tested?

Existing tests

# Are there any user-facing changes?

If anyone used `SetVariable` directly they will now have to update their code slightly

### CODE REVIEW

File: datafusion/core/src/physical_plan/planner.rs [hunk 1]
```diff
@@ -1157,21 +1157,11 @@ impl DefaultPhysicalPlanner {
                         "Unsupported logical plan: Dml".to_string(),
                     ))
                 }
-                LogicalPlan::TransactionStart(_) => {
+                LogicalPlan::Statement(statement) => {
```
Comment (alamb) [inline datafusion/core/src/physical_plan/planner.rs:1160]: this is the key difference -- collapse several LogicalPlan variants into one that all have the same behavior

File: datafusion/expr/src/logical_plan/mod.rs [hunk 2]
```diff
@@ -19,16 +19,20 @@ pub mod builder;
...
 pub use builder::{table_scan, LogicalPlanBuilder};
 pub use plan::{
     Aggregate, Analyze, CreateCatalog, CreateCatalogSchema, CreateExternalTable,
-    CreateMemoryTable, CreateView, CrossJoin, DescribeTable, Distinct, DmlStatement,
-    DropTable, DropView, EmptyRelation, Explain, Extension, Filter, Join, JoinConstraint,
-    JoinType, Limit, LogicalPlan, Partitioning, PlanType, Prepare, Projection,
-    Repartition, SetVariable, Sort, StringifiedPlan, Subquery, SubqueryAlias, TableScan,
-    ToStringifiedPlan, TransactionAccessMode, TransactionConclusion, TransactionEnd,
-    TransactionIsolationLevel, TransactionStart, Union, Unnest, Values, Window, WriteOp,
+    CreateMemoryTable, CreateView, CrossJoin, DescribeTable, Distinct, DropTable,
```
Comment (alamb) [inline datafusion/expr/src/logical_plan/mod.rs:27]: this makes the imports consistent

File: datafusion/expr/src/logical_plan/plan.rs [hunk 3]
```diff
@@ -1226,18 +1200,6 @@ pub struct DropView {
     pub schema: DFSchemaRef,
 }
 
-/// Set a Variable's value -- value in
```
Comment (alamb) [inline datafusion/expr/src/logical_plan/plan.rs:1229]: moved into `statement.rs`

File: datafusion/expr/src/logical_plan/plan.rs [hunk 4]
```diff
@@ -1559,83 +1521,6 @@ impl Hash for CreateExternalTable {
     }
 }
 
-#[derive(Clone, PartialEq, Eq, Hash)]
```
Comment (alamb) [inline datafusion/expr/src/logical_plan/plan.rs:1562]: Moved to statement.rs

**Review by crepererum — APPROVED** [review]
While I think we should ideally peruse https://github.com/apache/arrow-datafusion/issues/3349#issuecomment-1370092784 , this PR at least moves the code base into the right direction :+1:

**Pull request merged**